src/commands/report.ts

```diff
@@ -19,32 +27,51 @@ export async function execute(interaction: CommandInteraction) {
 
   try {
     const { guildId, user, options } = interaction;
-    const url = options.getString("url", true);
-    const escapedUrl = url.replace(".", "(dot)");
-
-    // submit report
-    const response = await axios.post(
-      `${env.CHAINPATROL_API_URL}/api/v2/report/create`,
-      {
-        discordGuildId: guildId,
-        title: "Discord Report",
-        description: `reported by discord user ${user.username} , Discord ID: ${user.id}`,
-        contactInfo: `discord user ${user.username} , Discord ID: ${user.id}`,
-        assets: [
+    const modal = generateModal(user, options);
+
+    // Show the modal to the user
+    await interaction.showModal(modal);
+
+    // extract data from modal
+    interaction
+      .awaitModalSubmit({ time: 60000 })
+      .then(async (modalInteraction) => {
```

Comment:
Can we `await` this promise instead of chaining with `.then()` and `.catch()`?

Reply:
https://github.com/chainpatrol/discord-bot/pull/23/commits/0d41ddac98242f6dbc5da7c1fcacd6ea1c3168a9